src/file-manager/fm-desktop-icon-view.c

```diff
@@ -373,6 +392,9 @@ fm_desktop_icon_view_handle_middle_click (CajaIconContainer *icon_container,
 static void
 unrealized_callback (GtkWidget *widget, FMDesktopIconView *desktop_icon_view)
 {
+    if (!GDK_IS_X11_DISPLAY (gdk_display_get_default))
+        return;
+
```

Comment:
The test is wrong (should again be `gdk_display_get_default ()`), but more importantly the whole function should only be called on X11 if it's not used anywhere else.

Or if we want to keep the realize/unrealize pair still, maybe wrap this in an `if () { }` instead to suggest there could be potential?  Either way is fine though, but you need to fix the test.

Reply:
Just fixed that one, it was a cut and paste from the same problem elsewhere fixed yesterday.

Reply:
Fixing this one also stopped the segfault on calling `caja --quit` in wayland with the desktop showing, which was exactly the condition this was intended to stop but it didn't work when I screwed up the conditional